controller/index.js

```diff
@@ -1,1019 +1,1054 @@
-                res.contentType('text/csv');
-              } else if ( req.params.format == 'zip'){
-                res.contentType('application/octet-stream');
-              }
+  /**
+   * Creates a clean and normalized name to be used in things like files on disk
+   *
+   * @params {string} type - the type of the providers ("agol")
+   * @params {object} params - an object with an item id and layer number
+   *
+   * @returns {string} key
+   * @private
+   */
+  controller._createName = function (info, key, format) {
+    format = format.replace('geojson', 'json')
 
-              if ( result.substr(0,4) == 'http' ){
-                // Proxy to s3 urls allows us to not show the URL 
-                https.get(result, function(proxyRes) {
-                  proxyRes.pipe(res);
-                });
-              } else {
-                res.sendfile(result);
-              }
-            }
-          }
-        });
-      } else {
-        res.status(400).send( 'Could not create export, missing data' );
+    var name = (info && info.info) ? info.name || info.info.name || info.info.title : key
+    name = (name.length > 150) ? name.substr(0, 150) : name
+
+    var fileName = name + '.' + format
+    fileName = fileName.replace(/\/|,|&|\|/g, '').replace(/ /g, '_').replace(/\(|\)/g, '')
+
+    return fileName
+  }
+
+  /**
+   * Creates a clean file path for files exports
+   *
+   * @params {string} type - the type of the providers ("agol")
+   * @params {object} params - an object with an item id and layer number
+   *
+   * @returns {string} key
+   * @private
+   */
+  controller._createFilePath = function (key, params) {
+    var dir = params.item + '_' + (params.layer || 0)
+    return ['files', dir, key].join('/')
+  }
+
+  /**
+   * Create key used to query a table in the cache
+   *
+   * @params {string} type - the type of the providers ("agol")
+   * @params {object} params - an object with an item id and layer number
+   *
+   * @returns {string} key
+   * @private
+   */
+  controller._requestNewFile = function (params) {
+    var res = params.res
+    var itemJson = params.itemJson
+
+    if (params.err) {
+      return res.status(params.err.code || 400).send(params.err.error || params.err)
+    }
+
+    // flatten the data from an array to sep objects/arrays
+    var itemData
+    var itemFeatures
+    if (itemJson && itemJson.data && itemJson.data[0]) {
+      itemData = itemJson.data[0]
+    }
+
+    if (itemData && itemData.features) {
+      itemFeatures = itemData.features
+    }
+
+    if (!itemFeatures || !itemFeatures.length) {
+      return res.status(404).send('No features exist for the requested FeatureService layer')
+    }
+
+    // this logic sure does suck...
+    var name = (itemJson && itemJson.data && itemJson.data[0] && (itemJson.data[0].name || (itemJson.data[0].info && itemJson.data[0].info.name))) ? itemJson.data[0].name || itemJson.data[0].info.name : itemJson.name || itemJson.title
```

Comment:
more specifically - this seems to be prone to syntax errors and logic confusion. Probably better to split these out for readability + maintainability.